lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileIterator.java

```diff
@@ -54,7 +53,9 @@ public FileEntry next()
             return null;
         }
         URIStatus fileStatus = files.next();
-        Location location = convertToLocation(fileStatus.getPath(), mountRoot);
+        String filePath = fileStatus.getPath();
+        String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+        Location location = pathLocation.appendSuffix("/" + fileName);
```

Comment:
This change may lead to failure of some other cases and some tests is not able to pass. But the bug does exist. Still need to find a way to fix it.

Reply:
What would be a more reasonable solution?

Reply:
Error:  Tests run: 24, Failures: 3, Errors: 0, Skipped: 4, Time elapsed: 62.74 s <<< FAILURE! -- in io.trino.filesystem.alluxio.TestAlluxioFileSystem
2024-10-17T02:10:31.425-0600	INFO	main	io.trino.testing.containers.junit.ReportLeakedContainers	ReportLeakedContainers disabled
Error:  io.trino.filesystem.alluxio.TestAlluxioFileSystem.testPaths -- Time elapsed: 0.051 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 35L
 but was: 34L
	at io.trino.filesystem.AbstractTestTrinoFileSystem.listPath(AbstractTestTrinoFileSystem.java:1370)
	at io.trino.filesystem.AbstractTestTrinoFileSystem.testPathHierarchical(AbstractTestTrinoFileSystem.java:763)
	at io.trino.filesystem.AbstractTestTrinoFileSystem.testPaths(AbstractTestTrinoFileSystem.java:739)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)

Error:  io.trino.filesystem.alluxio.TestAlluxioFileSystem.testListFiles -- Time elapsed: 0.171 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 46L
 but was: 59L
	at io.trino.filesystem.AbstractTestTrinoFileSystem.listPath(AbstractTestTrinoFileSystem.java:1370)
	at io.trino.filesystem.AbstractTestTrinoFileSystem.testListFiles(AbstractTestTrinoFileSystem.java:1070)
	at io.trino.filesystem.AbstractTestTrinoFileSystem.testListFiles(AbstractTestTrinoFileSystem.java:1001)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)

Error:  io.trino.filesystem.alluxio.TestAlluxioFileSystem.testListLexicographicalOrder -- Time elapsed: 1.436 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 44L
 but was: 43L
	at io.trino.filesystem.AbstractTestTrinoFileSystem.listPath(AbstractTestTrinoFileSystem.java:1370)
	at io.trino.filesystem.AbstractTestTrinoFileSystem.testListLexicographicalOrder(AbstractTestTrinoFileSystem.java:1207)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1458)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2034)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:189)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    TestAlluxioFileSystem>AbstractTestTrinoFileSystem.testListFiles:1001->AbstractTestTrinoFileSystem.testListFiles:1070->AbstractTestTrinoFileSystem.listPath:1370 
expected: 46L
 but was: 59L
Error:    TestAlluxioFileSystem>AbstractTestTrinoFileSystem.testListLexicographicalOrder:1207->AbstractTestTrinoFileSystem.listPath:1370 
expected: 44L
 but was: 43L
Error:    TestAlluxioFileSystem>AbstractTestTrinoFileSystem.testPaths:739->AbstractTestTrinoFileSystem.testPathHierarchical:763->AbstractTestTrinoFileSystem.listPath:1370 
expected: 35L
 but was: 34L
[INFO] 
Error:  Tests run: 27, Failures: 3, Errors: 0, Skipped: 4

Reply:
@awkwardd  Add the following `extractSchema` method to `AlluxioUtils`, and update the `next()` method in `AlluxioFileIterator` can pass the test successfully. Please help update the code in this PR.
```
    public static Location extractSchema(Location location) {
        String path = location.toString();
        if (path.startsWith("alluxio://")) {
            int index = path.indexOf("/", "alluxio://".length());
            if (index != -1) {
                String schema = path.substring(0, index);
                return Location.of(schema);
            } else {
                return location;
            }
        }
        return location;
    }
```

```
   public FileEntry next()
            throws IOException
    {
        if (!hasNext()) {
            return null;
        }
        URIStatus fileStatus = files.next();
        String path = fileStatus.getPath();
        if (path.startsWith("/")) {
            path = path.substring(1);
        }
        Location schema = extractSchema(dirLocation);
        Location fileLocation = schema.appendSuffix(path);
        return new FileEntry(
                fileLocation,
                fileStatus.getLength(),
                Instant.ofEpochMilli(fileStatus.getLastModificationTimeMs()),
                Optional.empty());
    }
```